components/project/traces/trace-filter.tsx

```diff
@@ -0,0 +1,438 @@
+                <Checkbox
+                  checked={selectedFilters.includes(method)}
+                  onClick={() => handleFilterChange(method)}
+                  value={method}
+                />
+
+                <label className="ml-2">{method}</label>
+              </div>
+            ))}
+        </div>
+        <div>
+          <h4
+            onClick={() => setShowPinecone(!showPinecone)}
+            className="cursor-pointer flex items-center"
+          >
+            Pinecone
+            <ChevronDownIcon className="ml-2 h-4 w-4" />
+          </h4>
+          {showPinecone &&
+            Object.values(PineconeMethods).map((method) => (
+              <div key={method} className="flex items-center">
+                <Checkbox
+                  checked={selectedFilters.includes(method)}
+                  onClick={() => handleFilterChange(method)}
+                  value={method}
+                />
+
+                <label className="ml-2">{method}</label>
+              </div>
+            ))}
+        </div>
+        <div>
+          <h4 className="mt-4">Attributes</h4>
+          {advancedFilters.map((filter, index) => (
+            <div key={index} className="flex items-center mt-2 space-x-2">
+              <AttributesCombobox
+                initialAttribute={filter.attribute}
+                setSelectedAttribute={(attribute) =>
+                  updateAdvancedFilter(index, "attribute", attribute)
+                }
+              />
+              <OperatorCombox
+                initialOperator={filter.operator}
+                setSelectedOperator={(operator) =>
+                  updateAdvancedFilter(index, "operator", operator)
+                }
+              />
+              <Input
+                placeholder="Value"
+                value={filter.value}
+                onChange={(e) =>
+                  updateAdvancedFilter(index, "value", e.target.value)
+                }
+                className="mr-2"
+              />
+              <Button onClick={() => removeAdvancedFilter(index)}>
+                Remove
+              </Button>
+            </div>
+          ))}
+          <Button onClick={addAdvancedFilter} className="mt-2">
+            Add Attribute
+          </Button>
+        </div>
+        <DialogFooter>
+          <Button onClick={onClose}>Cancel</Button>
+          <Button onClick={applyFilters} color="primary">
+            Apply Filters
+          </Button>
+        </DialogFooter>
+      </DialogContent>
+    </Dialog>
+  );
+}
+
+export function AttributesCombobox({
+  setSelectedAttribute,
+  initialAttribute,
+}: {
+  setSelectedAttribute: (attribute: string) => void;
+  initialAttribute: string;
+}) {
+  const [open, setOpen] = useState(false);
+  const [selectedAttribute, setSelectedAttributeState] = useState(
+    initialAttribute || ""
+  );
+  const [searchQuery, setSearchQuery] = useState("");
+
+  const attributes = [
```

Comment:
we should import these attributes from trace attributes, cos otherwise we will have to constantly maintain this list.